src/circuit/gadget/sinsemilla.rs

```diff
@@ -207,7 +207,13 @@ where
                     .map(|byte| byte.iter().rev().fold(0u8, |acc, bit| acc * 2 + *bit as u8))
                     .collect()
             });
-            bytes.map(|bytes| C::Base::from_bytes(&bytes.try_into().unwrap()).unwrap())
+            bytes.map(|bytes| {
+                let mut repr = <C::Base as PrimeField>::Repr::default();
+                // The above code assumes the byte representation is 256 bits.
```

Comment:
This implies that we need an API that goes from bits to field elements (regardless of the field's chosen endianness of its byte encoding).

```suggestion
                // The above code assumes the byte representation is 256 bits.
                // The above code also assumes that the encoding is little-endian
                // (since `FieldExt::from_bytes()` was explicitly defined as from
                // little-endian bytes), and will produce the wrong result if the
                // encoding is big-endian.
```